session/session_test.go

```diff
@@ -48,11 +52,75 @@ func TestReceiveEventWhenSubscribing(t *testing.T) {
 		if event != SessionDisconnected {
 			t.Error("Expected to receive disconnected: ", event)
 		}
+	case <-time.After(5 * time.Second):
+		t.Error("Failed to receive event")
+	}
 
-		if <-events != SessionReconnected {
+	select {
+	case event := <-events:
+		if event != SessionReconnected {
 			t.Error("Expected to receive reconnected: ", event)
 		}
-	case <-time.After(100 * time.Millisecond):
+	case <-time.After(5 * time.Second):
 		t.Error("Failed to receive event")
 	}
 }
+
+func TestResumeZKSessionWithValidSession(t *testing.T) {
+	proxy := CreateProxy(t)
+	defer proxy.Delete()
+
+	store, err := NewZKSession("localhost:27445", 200*time.Millisecond, nil)
+	if err != nil {
+		t.Error("Failed to connect to Zookeeper: ", err)
+	}
+
+	events := make(chan ZKSessionEvent)
+	store.Subscribe(events)
+
+	existingClientId, err := store.ClientId().Save()
+	if err != nil {
+		t.Error("Failed to save clientId: ", err)
+	}
+
+	clientId, err := zookeeper.LoadClientId(existingClientId)
+	if err != nil {
+		t.Error("Error loading clientId: ", err)
+	}
+
+	resumeStore, err := ResumeZKSession("localhost:27445", 200*time.Millisecond, nil, clientId)
+	if err != nil {
+		t.Error("Failed to resume session with Zookeeper: ", err)
+	}
+	defer resumeStore.Close()
+	store.Close()
```

Comment:
I don't think you should need to do this. The `store` should be closed when it's reconnected.